tardis/apps/publication_workflow/components/Forms/SelectDatasetForm.jsx

```diff
@@ -0,0 +1,215 @@
+    const selectedOptionsArray = [].slice.call(event.target.selectedOptions);
+    let currentlySelectedDatasetList = [];
+    selectedOptionsArray
+      .map(item => (currentlySelectedDatasetList
+        .push({
+          experiment: selectedExperiment.title,
+          experiment_id: selectedExperiment.id,
+          dataset: {
+            id: item.value,
+            description: item.label.slice(0, item.label.length),
+          },
+        })));
+    currentlySelectedDatasetList = selectedDatasetList.concat(currentlySelectedDatasetList);
+    setSelectedDatasetList(currentlySelectedDatasetList);
+    // run field validation
+    formik.validateField('selectedDatasets');
+  };
+
+  useEffect(() => {
+    fetchExperimentList().then((data) => {
+      const expListArray = [];
+      expListArray.push({ id: '-1', title: 'Select experiment' });
+      data.map(item => expListArray.push(item));
+      setExpList(expListArray);
+      formik.setFieldValue('selectedDatasets', selectedDatasetList, false);
+    });
+  }, [selectedDatasetList]);
+
+  const handleDatasetDelete = (e, item) => {
+    // remove this item from selectedDatasetList
+    const newSelectedDatasetList = selectedDatasetList.filter(it => it !== item);
+    setSelectedDatasetList(newSelectedDatasetList);
+  };
+  const onFilter = ({ target: { value } }) => {
+    const filter = value.trim().toLowerCase();
+    if (!filter) {
+      // GET ORIGINAL LIST
+      fetchDatasetsForExperiment(selectedExperiment.id).then(result => setDatasetList(result));
+    }
+    const filteredData = datasetList
+      .filter(({ description }) => description.toLowerCase().includes(filter));
+    setDatasetList(filteredData);
+  };
+  return (
+    <>
+      <p>
+        <strong>Select your data!</strong>
+        {' '}
+        Below is all the data
+        you&apos;ve generated. Select the datasets you want to include in
+        this publication.
+      </p>
+      <Card>
+        <Card.Body>
+          <Form.Group className="mb-3" controlId="formGroupTitle">
+            <Form.Label>Publication Title</Form.Label>
+            <Form.Control
+              type="text"
+              placeholder="concise description"
+              name="publicationTitle"
+              onChange={formik.handleChange}
+              value={formik.values.publicationTitle}
+              isValid={formik.touched.publicationTitle && !formik.errors.publicationTitle}
+              isInvalid={!!formik.errors.publicationTitle}
+            />
+            <Form.Control.Feedback type="invalid">
+              {formik.errors.publicationTitle}
+            </Form.Control.Feedback>
+          </Form.Group>
+          <Form.Group className="mb-3" controlId="formGroupDescription">
+            <Form.Label>Description</Form.Label>
+            <Form.Control
+              as="textarea"
+              name="publicationDescription"
+              rows={3}
+              placeholder="Describe your publication..."
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 28 spaces but found 14.](https://app.codacy.com/gh/mytardis/mytardis/pullRequest?prid=7933947)
```suggestion
                            placeholder="Describe your publication..."
```